File: spec/python/test_bcd_user_type_be.py
import unittest

from bcd_user_type_be import BcdUserTypeBe

class TestBcdUserTypeBe(unittest.TestCase):
    def test_bcd_user_type_be(self):
        r = BcdUserTypeBe.from_file("src/bcd_user_type_be.bin")

        self.assertEqual(r.ltr.as_int, 12345678)
        self.assertEqual(r.ltr.as_str, "12345678")
        self.assertEqual(r.rtl.as_int, 87654321)
        self.assertEqual(r.rtl.as_str, "87654321")
        self.assertEqual(r.leading_zero_ltr.as_int, 123456)
        self.assertEqual(r.leading_zero_ltr.as_str, "00123456")

File: spec/javascript/test_bcd_user_type_be.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('BcdUserTypeBe', 'src/bcd_user_type_be.bin', function(r) {
  assert.equal(r.ltr.asInt, 12345678);
  assert.equal(r.ltr.asStr, "12345678");
  assert.equal(r.rtl.asInt, 87654321);
  assert.equal(r.rtl.asStr, "87654321");
  assert.equal(r.leadingZeroLtr.asInt, 123456);
  assert.equal(r.leadingZeroLtr.asStr, "00123456");
});
